packages/office-ui-fabric-react/src/common/shallowUntilTarget.ts

```diff
@@ -0,0 +1,64 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+
+/**
+ * Duplicated enzyme's ShallowRendererProps
+ *
+ * @internal
+ */
+export interface IShallowRendererProps {
+  lifecycleExperimental?: boolean;
+  disableLifecycleMethods?: boolean;
+}
+
+/**
+ * ShallowUntilTarget Interface
+ *
+ * @internal
+ */
+export interface IShallowUntilTarget {
+  maxTries: number;
+  shallowOptions: IShallowRendererProps;
+}
+
+/**
+ * An extention of enzyme's shallow function which will fail to work
+ * with decorated components and/or components using the styled() function.
+ * This function allows you to pass a 'target' component (e.g. ComponentBase)
+ * and keep running shallow on each child component till a match is found.
+ *
+ * @public
+ */
+export function shallowUntilTarget<P, S>(
+  componentInstance: React.ReactElement<P>,
+  TargetComponent: string,
+  options: IShallowUntilTarget = {
+    maxTries: 10,
+    shallowOptions: {}
+  }
+): ShallowWrapper {
+  const { maxTries, shallowOptions } = options;
+
+  let root = shallow<P, S>(componentInstance, shallowOptions);
+
+  if (typeof root.type() === 'string') {
+    // If type() is a string then it's a DOM Node.
+    // If it were wrapped, it would be a React component.
+    throw new Error('Cannot unwrap this component because it is not wrapped');
```

Comment:
I think that we shouldn't throw in this case. I should be able to use this function anywhere that `shallow` can be used. If the TargetComponent is a div, and the root is a div then that's a fine case. If this should only be used in the case of a decorated component, then the name should reflect that. Something like `shallowDecoratedComponent`